src/parser/jobs/rdm/modules/MovementSkills.js

```diff
@@ -0,0 +1,205 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import React, {Fragment} from 'react'
+import {Accordion, Message} from 'semantic-ui-react'
+
+import ACTIONS from 'data/ACTIONS'
+import Rotation from 'components/ui/Rotation'
+import Module from 'parser/core/Module'
+import {Rule, Requirement} from 'parser/core/modules/Checklist'
+import {ActionLink} from 'components/ui/DbLink'
+import {getDataBy} from 'data'
+
+const CACS_PER_MANAFICATION = 3
+const DISPS_PER_MANAFICATION = 3
+const ENGAGEMENTS_PER_MANAFICATION = 4
+//This is due to the fact that manafication happens after your first cac/disp
+const GRACE_FOR_PULL = 2
+
+export default class MovementSkills extends Module {
+	static handle = 'movementskills'
+	static title = t('rdm.movementskills.title')`Movement Skills`
+	static dependencies = [
+		'checklist',
+	]
+
+	_history = []
+	_last_manafic = {
+		timestamp: 0,
+		cac: 0,
+		disp: 0,
+		engagement: 0,
+		events: [],
+	}
+
+	constructor(...args) {
+		super(...args)
+		//Default Target to hit
+		this.target = 95
+		this.addHook('cast', {by: 'player'}, this._onCast)
+		//this._history.push(this._last_manafic)
+		this.description = <Trans id="rdm.movementskills.description">Your movement skills are primarily used for damage, make sure you get 3 <ActionLink {...getDataBy(ACTIONS, 'id', ACTIONS.CORPS_A_CORPS.id)} /> and either 3 <ActionLink {...getDataBy(ACTIONS, 'id', ACTIONS.DISPLACEMENT.id)} /> or 4 <ActionLink {...getDataBy(ACTIONS, 'id', ACTIONS.ENGAGEMENT.id)} /> per <ActionLink {...getDataBy(ACTIONS, 'id', ACTIONS.MANAFICATION.id)} /> </Trans>
+		this.addHook('complete', this._onComplete)
+		//So we don't get negative numbers on the final panel display
+		this._last_manafic.timestamp = this.parser.fight.start_time
+	}
+
+	_onCast(event) {
+		const abilityID = event.ability.guid
+		switch (abilityID) {
+		case ACTIONS.MANAFICATION.id:
+			//push the previous manafication instance
+			this._history.push(this._last_manafic)
+			//setup the new instance, since we're unlikely to end the fight on manafication push
+			this._last_manafic = {
+				timestamp: event.timestamp,
+				cac: 0,
+				disp: 0,
+				engagement: 0,
+				events: [],
+			}
+			break
```

Comment:
if you `return` here, you can move the `events.push` outside the switch so it's not duped everywhere

Reply:
I only want to capture disp, engage, and cac and absolutely no other events; wouldn't moving it outside the switch cause me to capture every single event?

Reply:
If you only want those, should probably filter the hook as such - sorry for missing that.

can also utilise the default case to return, but that's pretty ugly.

would be good if you could avoid the conditionals, too